src/mutation/RelayMutation.js

```diff
@@ -50,12 +50,29 @@ class RelayMutation<Tp: Object> {
     route: RelayMetaRoute
   ) => Variables;
 
-  props: Tp;
+  _context: RelayContext;
   _didShowFakeDataWarning: boolean;
+  _unresolvedProps: Tp;
+  props: Tp;
 
   constructor(props: Tp) {
-    this._didShowFakeDataWarning = false;
-    this._resolveProps(props);
+    this._unresolvedProps = props;
+  }
+
+  /**
+   * @internal
+   */
+  bindContext(context: RelayContext): void {
+    if (!this._context) {
+      this._context = context;
+      this._resolveProps();
```

Comment:
If you have time that would be great, but we can move forward with the PR without it (most mutations don't rely on the removed behavior).